docs/content/functioneel/zaaktypen_api.md

```diff
@@ -0,0 +1,182 @@
+ZAAKTYPEN API
+===============================================
+**FUNCTIONELE DOCUMENTATIE**
+
+| **Aspect**      | **Beschrijving**                                                                                                                                                                                  |
+|-----------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Doel            | Het kunnen aanspreken van een voorziening voor het onderhouden en raadplegen van zaaktypen en de zaaktypecatalogi (ZTC) waarvan zij deel uit maken. |
```

Comment:
Ik denk toch dat 'onderhouden' niet het juiste woord is. Dat klinkt al functionaliteit voor het onderhouden terwijl het alleen om het vastleggen van gegevens gaat. Voorstel: '... voor het vastleggen en raadplegen van (gegevens van) zaaktypen ...'.

Reply:
Op dit moment is de API ook enkel-lezen, dus ik zou het zelfs durven beperken tot '... voor het raadplegen van zaaktypen...`

Reply:
@ArjanKloosterboer vertical slicing! ;) Ik denk dat het belangrijk is dat we de functionele documentatie in sync houden met de huidige stand van zaken. Dat dat muteren er moet komen klopt 100%, maar daar moet qua techniek wel even het een en ander voor gebeuren _en_ er moet een userstory voor komen.

Momenteel kan er gemuteerd worden in de admin-interface van het ZTC, buiten de API om.

Reply:
Ik stel voor de functionele documentatie (ook) als ontwerp te beschouwen en, indien van toepassing, daarin duidelijk te maken welke delen van het ontwerp nog niet in de API gerealiseerd zijn. Anders levert die documentatie direct vragen op: 'waarom kan dat niet'?

Reply:
Hmmm, zo worden er via functionele documentatie opeens functies of constructies bij bedacht die niet persé uit User Stories komen. Zorgt ook voor meer inconsistentie. Dat is een niet wenselijke situatie en gaat in tegen het proces en de DoD.

Reply:
Dan is er bij het uitwerken en realiseren van de userstory's toch iets mis gegaan. Je kunt alleen gegevens raadplegen als er functionaliteit is om deze te beheren. Dus in de userstory's waarin zaaktypegegevens geraadpleegd worden, had ook muteerfunctionaliteit gebouwd moeten worden op die gegevens. 
@ehotting, hoe kijk jij hier tegen aan als productowner?

Reply:
Ik ben het wel eens met Sergei: de muteerfunctionaliteit moet ingebouwd worden op basis van een eigen user story. Anders heeft m.i. het werken volgens userstories geen zin en kun je beter een complete REST API op alle resources in het RGBZ bouwen en achter kijken of alle userstories met die API in te vullen zijn.
Precies wat Joeri dan aangeeft gaat dat tegen het proces en de DoD in.

Reply:
Ik sluit aan bij Joeri en Michiel - als je alles op voorhand gaat uitdenken en neerschrijven in functionele documentatie, dan:

* gaat het heel erg op verkapte waterval lijken
* is de functionele documentatie helemaal niet in sync met de functionaliteit

Ik dacht dat het de bedoeling was om functionele documentatie samen met API spec en referentieimplementatie op basis van een user story bij te werken, en dat is een geheel, 1 slice van functionaliteit.

Reply:
Zit wat in. Maar, volgens mij ontbreken de muteer-operaties. 

> Je kunt alleen gegevens raadplegen als er functionaliteit is om deze er in te stoppen. Dus in de userstory's waarin zaaktypegegevens geraadpleegd worden, had ook muteerfunctionaliteit gebouwd moeten worden op die gegevens.

Reply:
Dan moet daar een userstory voor komen. De documentatie beschrijft wat er nu is.